### DISCUSSION TITLE
zeromq

### DISCUSSION THREAD

**Discussion by VWY-github** [BehaviorTree/BehaviorTree.CPP#785] [Q&A]
Is there documentation about the design decision to use zeromq?
Is it an optional dependency?
Where is zeromq used?

**Comment by nerudaj** [comment]
Based on CMakeLists.txt, it is required for connecting to Groot (I assume for live debugging). It seems to be optional if you don't need that functionality (I've been able to build randomly chosen examples just fine without this and without SQLite that can also be disabled through configure flag).
